lib/facility.rb

```diff
@@ -1,14 +1,69 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_data)
+    @name = facility_data[:name]
+    @address = facility_data[:address]
+    @phone = facility_data[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  end
+
+  def add_service(service)
+    unless @services.include?(service)
+      @services << service
+    end
+  end
+
+  def register_vehicle(model)
+    if @services.include?('Vehicle Registration')
+      unless registered_vehicles.include?(model)
+        registered_vehicles << model
+        registration_date = Date.today
+    
+        if model.electric_vehicle?
+          @collected_fees += 200
+        elsif model.antique?
+          @collected_fees += 25
+        else
+          @collected_fees += 100
+        end
+      end
+    end
+    registered_vehicles
   end
```

Comment:
Line 30 will not set that vehicles registration date, but rather assign today's date into a variable called `registration_date`. I see in your vehicle class that you created a method called `set_registration_date`. You could use that here and just say `model.set_registration_date`.

Reply:
You could also do something similar for `plate_type`.